driver-core/src/main/com/mongodb/operation/ChangeStreamBatchCursor.java

```diff
@@ -34,19 +35,24 @@
     private final ReadBinding binding;
     private final ChangeStreamOperation<T> changeStreamOperation;
 
-    private BsonDocument resumeToken;
+    private BsonTimestamp initialStartAtOperationTime;
     private BatchCursor<RawBsonDocument> wrapped;
+    private BsonDocument resumeToken;
 
     ChangeStreamBatchCursor(final ChangeStreamOperation<T> changeStreamOperation,
                             final BatchCursor<RawBsonDocument> wrapped,
                             final ReadBinding binding) {
         this.changeStreamOperation = changeStreamOperation;
-        this.resumeToken = changeStreamOperation.getResumeToken();
+        this.initialStartAtOperationTime = changeStreamOperation.getStartAtOperationTime();
         if (changeStreamOperation.getStartAfter() != null) {
-            this.resumeToken = changeStreamOperation.getStartAfter();
+            resumeToken = changeStreamOperation.getStartAfter();
+        }
+        else if (changeStreamOperation.getResumeAfter() != null) {
+            resumeToken = changeStreamOperation.getResumeAfter();
         }
-        if (this.resumeToken == null) {
+        if (resumeToken == null) {
             changeStreamOperation.startOperationTimeForResume(binding.getSessionContext().getOperationTime());
+            resumeToken = wrapped.getPostBatchResumeToken();
```

Comment:
Sorry, not sure that I understand this line. Why do we save `postBatchResumeToken` only if no `startAfter` and `resumeAfter`?